public/app/features/plugins/admin/components/UpdateAllModal.tsx

```diff
@@ -0,0 +1,283 @@
+import { css } from '@emotion/css';
+import { useEffect, useMemo, useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { config } from '@grafana/runtime';
+import { Checkbox, ConfirmModal, EmptyState, Icon, Spinner, Tooltip, useStyles2 } from '@grafana/ui';
+import { t, Trans } from 'app/core/internationalization';
+
+import { useInstall, useInstallStatus } from '../state/hooks';
+import { CatalogPlugin } from '../types';
+
+type UpdateError = {
+  id: string;
+  message: string;
+};
+
+function getIcon({
+  id,
+  inProgress,
+  errorMap,
+  selectedPlugins,
+}: {
+  id: string;
+  inProgress: boolean;
+  errorMap: Map<string, UpdateError>;
+  selectedPlugins?: Set<string>;
+}) {
+  if (errorMap && errorMap.has(id)) {
+    return (
+      <Tooltip
+        content={`${t('plugins.catalog.update-all.error', 'Error updating plugin:')} ${errorMap.get(id)?.message}`}
+      >
+        <Icon size="xl" name="exclamation-circle" />
+      </Tooltip>
+    );
+  }
+  if (inProgress && selectedPlugins?.has(id)) {
+    return <Spinner />;
+  }
+  return '';
+}
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  table: css({
+    marginTop: theme.spacing(2),
+    width: '100%',
+    borderCollapse: 'collapse',
+  }),
+  tableRow: css({
+    borderBottom: `1px solid ${theme.colors.border.weak}`,
+    td: {
+      paddingRight: theme.spacing(1),
+    },
+  }),
+  icon: css({
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  }),
+  header: css({
+    textAlign: 'left',
+    padding: theme.spacing(1),
+    borderBottom: `2px solid ${theme.colors.border.strong}`,
+    th: {
+      paddingRight: theme.spacing(1),
+    },
+  }),
+  data: css({
+    padding: '10px',
+  }),
+  footer: css({
+    fontSize: theme.typography.bodySmall.fontSize,
+    marginTop: theme.spacing(3),
+  }),
+  noPluginsMessage: css({
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    height: '100%',
+  }),
+  tableContainer: css({
+    overflowY: 'auto',
+    overflowX: 'hidden',
+    height: theme.spacing(32),
+  }),
+  modalContainer: css({
+    height: theme.spacing(41),
+  }),
+});
+
+type ModalBodyProps = {
+  plugins: CatalogPlugin[];
+  inProgress: boolean;
+  selectedPlugins?: Set<string>;
+  onCheckboxChange: (id: string) => void;
+  errorMap: Map<string, UpdateError>;
+};
+
+const ModalBody = ({ plugins, inProgress, selectedPlugins, onCheckboxChange, errorMap }: ModalBodyProps) => {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div className={styles.modalContainer}>
+      {plugins.length === 0 ? (
+        <EmptyState
+          variant="completed"
+          message={t('plugins.catalog.update-all.all-plugins-updated', 'All plugins updated!')}
+        />
+      ) : (
+        <>
+          <div>
+            <Trans i18nKey="plugins.catalog.update-all.header">The following plugins have update available</Trans>
```

Comment:
The following plugins have update available 

Should be:
- The following plugins have an update available